src/Idris/DSL.hs

```diff
@@ -17,13 +17,30 @@ desugar :: SyntaxInfo -> IState -> PTerm -> PTerm
 desugar syn i t = let t' = expandDo (dsl_info syn) t in
                       t' -- addImpl i t'
 
+getNameStr :: Name -> PTerm
+getNameStr (UN nm)   = PConstant $ Str $ str nm
+getNameStr (NS nm _) = getNameStr nm
+getNameStr (MN _ nm) = PConstant $ Str $ str nm
```

Comment:
`MN` denotes a name chosen by the machine rather than by the user. It might be a name for a `_` pattern or for the binder in a non-dependent function type. It seems to be like a bad idea to lift these.